voice/src/jvmMain/kotlin/dev/kord/voice/udp/DefaultAudioPacketProvider.kt

```diff
@@ -8,14 +8,12 @@ import dev.kord.voice.io.MutableByteArrayCursor
 import dev.kord.voice.io.mutableCursor
 import dev.kord.voice.io.view
 
-public abstract class AudioPacketProvider(public val key: ByteArray, public val nonceStrategy: NonceStrategy) {
-    public abstract fun provide(sequence: UShort, timestamp: UInt, ssrc: UInt, data: ByteArray): ByteArrayView
-}
+@Suppress("FunctionName")
+public actual fun DefaultAudioPacketProvider(key: ByteArray, nonceStrategy: NonceStrategy) : AudioPacketProvider =
+    DefaultJvmAudioPacketProvider(key, nonceStrategy)
 
-private class CouldNotEncryptDataException(data: ByteArray) :
-    RuntimeException("Couldn't encrypt the following data: [${data.joinToString(", ")}]")
 
-public class DefaultAudioPacketProvider(key: ByteArray, nonceStrategy: NonceStrategy) :
+public class DefaultJvmAudioPacketProvider(key: ByteArray, nonceStrategy: NonceStrategy) :
```

Comment:
would be nice to keep the original name for compatibility

Reply:
> Conflicting overloads: public actual fun DefaultAudioPacketProvider(key: ByteArray, nonceStrategy: NonceStrategy): AudioPacketProvider defined in dev.kord.voice.udp in file DefaultAudioPacketProvider.kt, public constructor DefaultAudioPacketProvider(key: ByteArray, nonceStrategy: NonceStrategy) defined in dev.kord.voice.udp.DefaultAudioPacketProvider

Reply:
could rename the new `DefaultAudioPacketProvider` factory function to `AudioPacketProvider` instead

Reply:
We could, but that would change the meaning of the function, for the sake of keeping binary compatibility of an experimental part of the library, which @lost-illusi0n wants to rewrite anyway, so this will change anyw

Reply:
fine